src/auth_middleware.rs

```diff
@@ -47,7 +47,7 @@ impl PartialEq for AuthEndpoint {
         }
 
         for (i, path) in self_path.iter().enumerate() {
-            if other_path[i] != path.clone() && "*" != path {
+            if &other_path[i] != path && "*" != path {
```

Comment:
Clone incurs in penalties while taking a reference is almost free so this form is preferred.